drivers/at/at.c

```diff
@@ -264,69 +196,127 @@ int at_send_cmd(at_dev_t *dev, const char *command, uint32_t timeout)
     uart_write(dev->uart, (const uint8_t *)command, cmdlen);
     uart_write(dev->uart, (const uint8_t *)CONFIG_AT_SEND_EOL, AT_SEND_EOL_LEN);
 
-    if (!IS_ACTIVE(CONFIG_AT_SEND_SKIP_ECHO)) {
-        if (at_wait_bytes(dev, command, timeout)) {
-            return -1;
-        }
-
-        if (at_expect_bytes(dev, CONFIG_AT_SEND_EOL, timeout)) {
-            return -2;
-        }
+    if (IS_ACTIVE(CONFIG_AT_SEND_SKIP_ECHO)) {
+        return 0;
     }
-
-    return 0;
+    return wait_echo(dev, command, timeout);
 }
 
 void at_drain(at_dev_t *dev)
 {
     uint8_t _tmp[16];
     int res;
 
-#if IS_USED(MODULE_AT_URC_ISR)
-    dev->awaiting_response = true;
-#endif
-
     do {
         /* consider no character within 10ms "drained" */
         res = isrpipe_read_timeout(&dev->isrpipe, _tmp, sizeof(_tmp), 10000U);
     } while (res > 0);
+}
 
-#if IS_USED(MODULE_AT_URC_ISR)
-    dev->awaiting_response = false;
-#endif
+static bool is_eol(char p)
+{
+    return p == '\r' || p == '\n';
 }
 
-ssize_t at_send_cmd_get_resp(at_dev_t *dev, const char *command,
-                             char *resp_buf, size_t len, uint32_t timeout)
+static char *skip_leading_eol(char *line)
 {
-    ssize_t res;
+    while (is_eol(*line)) {
+        line++;
+    }
+    return line;
+}
 
-    at_drain(dev);
+static ssize_t trim_leading_eol(char *buf, size_t size)
```

Comment:
```suggestion
static size_t trim_leading_eol(char *buf, size_t strlen)
```
size of a buffer typically has the \0 included

why does this return "signed size_t" ?